### PR TITLE
Add tpch hash suite

### PR DESCRIPTION
**Pull request by metdos**
Example usage:

```
fab add.tpch:scale_factor=1,partition_type='hash'
```

### CODE REVIEW

**Comment by lithp** [comment]
Nice, looks good to me!

**Comment by onderkalaci** [comment]
Needs documentation 😄

File: tpch_2_13_0/tpch_create_hash_partitioned_tables.ddl [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+CREATE TABLE nation
+(
+    n_nationkey  INTEGER not null,
+    n_name       CHAR(25) not null,
+    n_regionkey  INTEGER not null,
+    n_comment    VARCHAR(152)
+);
+
+select create_distributed_table('nation', 'n_nationkey');
```
Comment (onderkalaci) [inline tpch_2_13_0/tpch_create_hash_partitioned_tables.ddl]: I think this should be `create_reference_table()` Same for `supplier `. @metdos Does that work?

**Comment by metdos** [comment]
> I think this should be create_reference_table() Same for supplier. @metdos Does that work?

Yeap that makes sense. Just, it would be conflicting with the settings, because we say 'hash' but also create reference tables.

We can provide 3 options;
1. Default, hash + reference tables.  
2. 'hash' -> all tables are hash partitioned
3. 'append' -> all tables are append partitioned

**Comment by lithp** [comment]
What does having multiple options save us? Why would we want only hash tables? And why do we want a mix of hash+references tables?

**Comment by metdos** [comment]
> What does having multiple options save us?

When you need them, it saves us from creating them manually. Same as why we need test automation.

> Why would we want only hash tables?

I would want to run some tests with only hash tables. You could not replicate this problem otherwise:
https://github.com/citusdata/citus/issues/542

> And why do we want a mix of hash+references tables?

Because some of them need to be reference tables in real life scenarios so it is the default. Other than that it is a good opportunity test them together.

**Comment by lithp** [comment]
Okay. It feels a little weird to provide all three options but if all all of them are independently useful I suppose it's not a lot of extra cruft.

**Pull request merged**